packages/app/src/common/hooks/use-analytics.ts

```diff
@@ -0,0 +1,31 @@
+import { useSelector } from 'react-redux';
+import { selectDecodedAuthRequest } from '@store/onboarding/selectors';
+import { doTrack as track, doTrackScreenChange } from '@common/track';
+import { doChangeScreen as changeScreen } from '@store/onboarding/actions';
+import { ScreenName } from '@store/onboarding/types';
+
+export const useAnalytics = () => {
+  const authRequest = useSelector(selectDecodedAuthRequest);
+
+  const getAppDetails = () => {
+    const appURL = authRequest ? new URL(authRequest?.redirect_uri) : null;
+    return {
+      appName: authRequest?.appDetails?.name,
+      appDomain: appURL?.host,
+    };
+  };
```

Comment:
We have useAppDetails hook already, should we use that here?

Reply:
Well, the app details come from the `authRequest`. So I was only using one thing. But, agreed that `useAppDetails` is cleaner.

Reply:
ah yes ok I see. Not to nit too much, but I think it would probs be cleaner to also include the app domain in the redux store and add it to the useAppDetails hook

Reply:
cool, yeah thats a good suggestion! added.